### PR TITLE
[INLONG-3383][Manager] Fix the null pointer caused by sink field not configured with source field

### LINKED ISSUE #3383: [Bug][Manager] Fix the null pointer caused by sink field not configured with source field
### What happened

A null pointer exception will occur  if the sink field is not configured with the source field information when distributing the sort configuration.

### What you expected to happen

It is expected that the sort configuration can be delivered normally, even if the sink field is not configured with source field information.

### How to reproduce

The sink field does not configure the source field information when distributing the sort configuration.

### Environment

_No response_

### InLong version

master

### InLong Component

InLong Manager

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by yunqingmoswu**
### Title Name: [INLONG-3383][Manager] Fix the null pointer caused by sink field not configured with source field

Fixes #3383 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve.*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/thirdparty/sort/util/FieldInfoUtils.java [hunk 1]
```diff
@@ -87,10 +87,11 @@
             FieldInfo sinkField = getFieldInfo(field.getFieldName(), field.getFieldType(),
                     field.getIsMetaField() == 1, field.getFieldFormat());
             sinkFields.add(sinkField);
-
-            FieldInfo sourceField = getFieldInfo(field.getSourceFieldName(),
-                    field.getSourceFieldType(), field.getIsMetaField() == 1, field.getFieldFormat());
-            mappingUnitList.add(new FieldMappingUnit(sourceField, sinkField));
+            if (StringUtils.isNotBlank(field.getSourceFieldName())) {
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/thirdparty/sort/util/FieldInfoUtils.java:90]: The source and sink fields cannot be empty here. If they are empty, it means that there is a problem with the configuration, and an exception should be thrown.

Comment (healchow) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/thirdparty/sort/util/FieldInfoUtils.java:90]: Perhaps we should do non-null checks when saving and modifying source and sink fields.

Comment (yunqingmoswu) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/thirdparty/sort/util/FieldInfoUtils.java:90]: Writing a data source like hive requires all table field information, but there are some field mapping scenarios, such as 8 fields on the user source side and 10 fields on the target side. At this time, all the fields on the target side need to be passed in. But only part of the field mapping can be passed in.

Comment (healchow) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/thirdparty/sort/util/FieldInfoUtils.java:90]: OK, then the Sort module needs to handle this scenario.

**Review by healchow — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**